Tests fail on Windows due to leftover \r when parsing documentation

Running tests on Windows 10, Python 3.8 results in 2 failures and 2 errors. The errors are due to #31.

One of the failures:
```
======================================================================
FAIL: TestCase('tests\\annotations.rnc')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test.py", line 32, in runTest
    self.assertBestEqual(expected, actual)
  File "test.py", line 8, in assertBestEqual
    self.assertMultiLineEqual(expected, actual)
AssertionError: '<?xm[1350 chars]tion\ncontinues on the next line</a:documentat[102 chars]mar>' != '<?xm[1350 chars]tion\r\ncontinues on the next line\r</a:docume[106 chars]mar>'
  [...snip...]
    <define name="bar">
-     <a:documentation>documentation for definition
+     <a:documentation>documentation for definition
?                                                  +
- continues on the next line</a:documentation>
+ </a:documentation>ext line
  [...snip...]
```
Both failures are because the computed rng documentation string contains a "\r" symbol. This appears to be due to: 
https://github.com/djc/rnc2rng/blob/master/rnc2rng/parser.py#L543
where the "\r" symbol (a newline character in Windows) is not removed from the end of `p[0].value`. I'm not sure how "\r" and "\n" are removed from all the other lines, but at least here, it is left over and should be removed to ensure proper matching in the test and proper multi-line parsing.

--- a/rnc2rng/parser.py
+++ b/rnc2rng/parser.py
@@ -4,13 +4,13 @@ import rply, sys, os
 from codecs import BOM_UTF16_BE, BOM_UTF16_LE
 if sys.version_info[0] < 3:
     from urllib2 import urlopen as _urlopen
-    from urlparse import urljoin
+    from urlparse import urljoin, urlparse
     from contextlib import closing
     def urlopen(f):
         return closing(_urlopen(f))
 else:
     from urllib.request import urlopen
-    from urllib.parse import urljoin
+    from urllib.parse import urljoin, urlparse
 
 KEYWORDS = set([
     'attribute', 'datatypes', 'default', 'div', 'element', 'empty', 'external',
@@ -213,7 +213,7 @@ def component_div(s, p):
 
 @pg.production('component : INCLUDE strlit opt-inherit opt-include-content')
 def component_include(s, p):
-    if ':' in s.path or ':' in p[1].value:  # it's a URL
+    if is_url(s.path) or is_url(p[1].value):  # it's a URL
         url = urljoin(s.path, p[1].value)
     else:
         url = os.path.join(s.path, p[1].value)
@@ -678,7 +678,7 @@ class State(object):
         self.fn = fn
         self.path = os.getcwd()
         if fn is not None:
-            self.path = os.path.dirname(os.path.abspath(fn)) if ':' not in fn else fn
+            self.path = os.path.dirname(os.path.abspath(fn)) if not is_url(fn) else fn
         self.lines = src.splitlines()
 
 if sys.version_info[0] < 3:
@@ -686,11 +686,15 @@ if sys.version_info[0] < 3:
 else:
     str_types = str, bytes
 
+def is_url(fn):
+    parse_result = urlparse(fn)
+    return parse_result.scheme in ('http', 'https', 'file')
+
 def parse(src=None, f=None):
     assert src is None or f is None
     if f is not None and isinstance(f, str_types):
         fn = f
-        if ':' in fn:
+        if is_url(fn):
             with urlopen(fn) as f:
                 bytes = f.read()
         else:

--- a/rnc2rng/parser.py
+++ b/rnc2rng/parser.py
@@ -540,7 +540,7 @@ def name_class_group(s, p):
 @pg.production('documentations : DOCUMENTATION documentations')
 def documentations_multi(s, p):
     cur = Node('DOCUMENTATION', None, []) if not p[1] else p[1][0]
-    cur.value.insert(0, p[0].value.lstrip('# '))
+    cur.value.insert(0, p[0].value.lstrip('# ').rstrip('\r'))
     return [cur]
 
 @pg.production('documentations : ')

--- a/test.py
+++ b/test.py
@@ -1,5 +1,12 @@
 import rnc2rng
 import unittest, os
+import sys
+if sys.version_info[0] < 3:
+    from urllib import pathname2url, url2pathname
+    from urlparse import urljoin, urlparse
+else:
+    from urllib.parse import urljoin, urlparse
+    from urllib.request import pathname2url, url2pathname
 
 
 class TestUtils(unittest.TestCase):
@@ -21,10 +28,11 @@ class FileTest(TestUtils):
         return 'TestCase(%r)' % self.fn
 
     def runTest(self):
-
         root = rnc2rng.load(self.fn)
         ref = self.fn.replace('.rnc', '.rng')
-        ref = ref[7:] if ref.startswith('file://') else ref
+        if ref.startswith('file:'):
+            parse_result = urlparse(ref)
+            ref = url2pathname(parse_result.path)
         with open(ref) as f:
             expected = f.read().rstrip()
 
@@ -50,7 +58,8 @@ def suite():
         fn = os.path.join('tests', fn)
         suite.addTest(FileTest(fn))
     # synthesize a test that reads its input from a URL
-    suite.addTest(FileTest('file://' + os.path.abspath('tests/include.rnc')))
+    url_test_path = os.path.abspath('tests/include.rnc')
+    suite.addTest(FileTest(urljoin('file:', pathname2url(url_test_path))))
     return suite
 
 

